### PR TITLE
Replaced Input for Typeahead for Expenses

### LINKED ISSUE #447: Change dropdown lists for expenses/rates to be typeahead lists
Some of the lists can get pretty long.  It shouldn't be too hard to use the typeahead to allow searching.  Everything else, like validation & calculations, should continue to work the same way.

ex: https://localhost:5001/expense/entry/38

### PR DESCRIPTION
**Pull request by wongband**
resolve #447

Todo:
Scott's second change request

### CODE REVIEW

File: Harvest.Web/ClientApp/src/App.tsx [hunk 1]
```diff
@@ -22,6 +22,7 @@ import { InvoiceListContainer } from "./Invoices/InvoiceListContainer";
 import { TicketCreate } from "./Tickets/TicketCreate";
 import { TicketsContainer } from "./Tickets/TicketsContainer";
 import { TicketDetailContainer } from "./Tickets/TicketDetailContainer";
+import 'react-bootstrap-typeahead/css/Typeahead.css';
```
Comment (srkirkland) [inline Harvest.Web/ClientApp/src/App.tsx:25]: why do we need this now where we didn't before?  It should already be in the index.tsx page and thus doesn't need to be declared twice.

**Review by srkirkland — CHANGES_REQUESTED** [review]
Two things I'd like to see changed, not exactly sure how to do the second. 
1. I'd like to have an easy way to clear out an option once selected, like they do in http://ericgio.github.io/react-bootstrap-typeahead/#controlling-selections.  Once it is cleared out, we need to reset the state as if nothing had been selected.  This is happening in the main branch with dropdowns but not here.
2. The options should be restricted to just what we offer.  It would be nice if people couldn't type random stuff into the box.

File: Harvest.Web/ClientApp/src/App.tsx (hunk 1 above)
Comment (wongband) [reply Harvest.Web/ClientApp/src/App.tsx:25]: Sorry didn't notice that. I will get rid of it.

**Review by jSylvestre — CHANGES_REQUESTED** [review]
On the quote page, this is not triggering the enable of the submit button

**Review by jSylvestre — CHANGES_REQUESTED** [review]
Save a quote, then open it back up and the info isn't populated:
image

**Review by jSylvestre — CHANGES_REQUESTED** [review]
Almost there.
The dropdown should extend out of the container:

image

**Review by jSylvestre — APPROVED** [review]
Works for me, but maybe wait for @srkirkland to take a look.

**Review by srkirkland — APPROVED** [review]

**Pull request merged**